src/Storages/MergeTree/MergeTreeWhereOptimizer.cpp

```diff
@@ -204,13 +210,35 @@ static void collectColumns(const RPNBuilderTreeNode & node, const NameSet & colu
 
     /// Do not account arguments of function "indexHint"
     /// because they won't be read from table.
-    if (function_node.getFunctionName() == "indexHint")
+    const auto function_name = function_node.getFunctionName();
+    if (function_name == "indexHint")
         return;
 
+    if (!ConditionSelectivityEstimator::atom_map.contains(function_name))
```

Comment:
This map is only used to estimate. I think we should use `KeyCondition::atom_map`